### ISSUE TITLE
Auto-uploading pictures fails sometimes

### ISSUE THREAD

**Issue by aronovgj** [nextcloud/android#85]
### Actual behaviour
- Only some pictures are uploaded (for example one of eight)
- Errors in Nextcloud log
- Client says upload failed
### Expected behaviour

Pictures should upload without problems
### Steps to reproduce
1. Set up auto-uploading pictures
2. Take some pictures

Happens sometimes. Couldn't see regularities. Sometimes it manages to upload everything
### Environment data

Android version:
OxygenOS 3.1.2

Device model: 
Oneplus 3

Stock or customized system:
stock,rooted/unlocked bootloader

Nextcloud app version:
1.1.0 RC

Nextcloud server version:
9.0.51
#### Nextcloud log (data/nextcloud.log)

```
{"reqId":"gdBhm6NXZUBzGH6BAGSl","remoteAddr":"xxx.xxx.xxx.xxx","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 400 expected filesize 1024000 got 819200\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\BadRequest\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/file.php(103): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->createFileChunked(Resource id #40)\\n#1 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/directory.php(134): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->put(Resource id #40)\\n#2 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1036): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Directory->createFile('IMG_20160701_00...', Resource id #40)\\n#3 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(523): Sabre\\\\DAV\\\\Server->createFile('pics\\\/IMG_201607...', Resource id #40, NULL)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#7 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(55): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/var\\\/www\\\/nextcloud\\\/remote.php(138): require_once('\\\/var\\\/www\\\/nextcl...')\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/file.php\",\"Line\":404,\"User\":\"me\"}","level":4,"time":"2016-07-01T00:54:33+02:00","method":"PUT","url":"\/remote.php\/webdav\/pics\/IMG_20160701_005407.jpg-chunking-8697-5-1","user":"me"}
{"reqId":"PSqH9LB\/2yACgOWaqgtN","remoteAddr":"xxx.xxx.xxx.xxx","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 400 expected filesize 1024000 got 466944\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\BadRequest\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/file.php(103): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->createFileChunked(Resource id #40)\\n#1 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/directory.php(134): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->put(Resource id #40)\\n#2 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1036): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Directory->createFile('IMG_20160701_00...', Resource id #40)\\n#3 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(523): Sabre\\\\DAV\\\\Server->createFile('pics\\\/IMG_201607...', Resource id #40, NULL)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#7 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(55): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/var\\\/www\\\/nextcloud\\\/remote.php(138): require_once('\\\/var\\\/www\\\/nextcl...')\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/file.php\",\"Line\":404,\"User\":\"me\"}","level":4,"time":"2016-07-01T00:54:36+02:00","method":"PUT","url":"\/remote.php\/webdav\/pics\/IMG_20160701_005410.jpg-chunking-7335-5-1","user":"me"}
{"reqId":"NkaZm9IwP6kn2TQARoFn","remoteAddr":"xxx.xxx.xxx.xxx","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 400 expected filesize 1024000 got 991232\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\BadRequest\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/file.php(103): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->createFileChunked(Resource id #40)\\n#1 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/directory.php(134): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->put(Resource id #40)\\n#2 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1036): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Directory->createFile('IMG_20160701_00...', Resource id #40)\\n#3 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(523): Sabre\\\\DAV\\\\Server->createFile('pics\\\/IMG_201607...', Resource id #40, NULL)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#7 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(55): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/var\\\/www\\\/nextcloud\\\/remote.php(138): require_once('\\\/var\\\/www\\\/nextcl...')\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/file.php\",\"Line\":404,\"User\":\"me\"}","level":4,"time":"2016-07-01T00:54:39+02:00","method":"PUT","url":"\/remote.php\/webdav\/pics\/IMG_20160701_005413.jpg-chunking-9223-6-1","user":"me"}
{"reqId":"hbN1PuU1kbdgadTgE6Aq","remoteAddr":"xxx.xxx.xxx.xxx","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 400 expected filesize 1024000 got 188416\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\BadRequest\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/file.php(103): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->createFileChunked(Resource id #40)\\n#1 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/directory.php(134): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->put(Resource id #40)\\n#2 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1036): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Directory->createFile('IMG_20160701_00...', Resource id #40)\\n#3 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(523): Sabre\\\\DAV\\\\Server->createFile('pics\\\/IMG_201607...', Resource id #40, NULL)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#7 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(55): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/var\\\/www\\\/nextcloud\\\/remote.php(138): require_once('\\\/var\\\/www\\\/nextcl...')\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/file.php\",\"Line\":404,\"User\":\"me\"}","level":4,"time":"2016-07-01T00:54:54+02:00","method":"PUT","url":"\/remote.php\/webdav\/pics\/IMG_20160701_005428.jpg-chunking-9104-5-1","user":"me"}
```

**Comment by tobiasKaminsky** [comment]
Are you suspending the app or swiping it out of the recent app list?

**Comment by aronovgj** [comment]
No. The app also is not kicked out by lmk when this happens

**Comment by aronovgj** [comment]
Also happens on a stable Wlan connection.

**Comment by tobiasKaminsky** [comment]
Can you try it with the demo server? demo.owncloud.org user/pass: test?

**Comment by aronovgj** [comment]
Done

![screenshot_20160702-111500](https://cloud.githubusercontent.com/assets/10271770/16539533/8300735e-4046-11e6-9833-a76939d2a00d.png)

**Comment by aronovgj** [comment]
Was that of any help? I just saw that only fatal issues are logged on the demo server.

**Comment by aronovgj** [comment]
That is a full log of uploading a file and failing. Please pay attention to the 404 which I did not have in the OP (maybe I simply did not see it). 
`404 File with name pics\\\/IMG_20160703_202123.jpg could not be located`


```
{"reqId":"YMM4H4cc04lKA99W\/nXx","remoteAddr":"xxx.xxx.xxx.xxx","app":"admin_audit","message":"Login attempt: \"me\"","level":1,"time":"2016-07-03T20:22:00+02:00","method":"HEAD","url":"\/remote.php\/webdav\/pics\/","user":"--"}
{"reqId":"YMM4H4cc04lKA99W\/nXx","remoteAddr":"xxx.xxx.xxx.xxx","app":"admin_audit","message":"Login successful: \"me\"","level":1,"time":"2016-07-03T20:22:00+02:00","method":"HEAD","url":"\/remote.php\/webdav\/pics\/","user":"me"}
{"reqId":"gD+pXy5LDmSxXtmh\/bF0","remoteAddr":"xxx.xxx.xxx.xxx","app":"admin_audit","message":"Login attempt: \"me\"","level":1,"time":"2016-07-03T20:22:00+02:00","method":"HEAD","url":"\/remote.php\/webdav\/pics\/IMG_20160703_202123.jpg","user":"--"}
{"reqId":"gD+pXy5LDmSxXtmh\/bF0","remoteAddr":"xxx.xxx.xxx.xxx","app":"admin_audit","message":"Login successful: \"me\"","level":1,"time":"2016-07-03T20:22:00+02:00","method":"HEAD","url":"\/remote.php\/webdav\/pics\/IMG_20160703_202123.jpg","user":"me"}
{"reqId":"gD+pXy5LDmSxXtmh\/bF0","remoteAddr":"xxx.xxx.xxx.xxx","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 404 File with name pics\\\/IMG_20160703_202123.jpg could not be located\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\NotFound\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(79): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\ObjectTree->getNodeForPath('pics\\\/IMG_201607...', 0)\\n#1 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpGet(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#2 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#3 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:GET', Array)\\n#4 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(251): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response), false)\\n#5 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpHead(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#6 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#7 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:HEAD', Array)\\n#8 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#9 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(55): Sabre\\\\DAV\\\\Server->exec()\\n#10 \\\/var\\\/www\\\/nextcloud\\\/remote.php(138): require_once('\\\/var\\\/www\{"reqId":"gD+pXy5LDmSxXtmh\/bF0","remoteAddr":"xxx.xxx.xxx.xxx","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 404 File with name pics\\\/IMG_20160703_202123.jpg could not be located\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\NotFound\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(79): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\ObjectTree->getNodeForPath('pics\\\/IMG_201607...', 0)\\n#1 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpGet(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#2 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#3 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:GET', Array)\\n#4 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(251): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response), false)\\n#5 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpHead(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#6 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#7 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:HEAD', Array)\\n#8 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#9 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(55): Sabre\\\\DAV\\\\Server->exec()\\n#10 \\\/var\\\/www\\\/nextcloud\\\/remote.php(138): require_once('\\\/var\\\/www\\\/nextcl...')\\n#11 {main}\",\"File\":\"\\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/objecttree.php\",\"Line\":168,\"User\":\"me\"}","level":0,"time":"2016-07-03T20:22:00+02:00","method":"HEAD","url":"\/remote.php\/webdav\/pics\/IMG_20160703_202123.jpg","user":"me"}
{"reqId":"+dz8shX\/6P6E79tpLi48","remoteAddr":"xxx.xxx.xxx.xxx","app":"admin_audit","message":"Login attempt: \"me\"","level":1,"time":"2016-07-03T20:22:00+02:00","method":"PUT","url":"\/remote.php\/webdav\/pics\/IMG_20160703_202123.jpg-chunking-1892-4-0","user":"--"}
{"reqId":"+dz8shX\/6P6E79tpLi48","remoteAddr":"xxx.xxx.xxx.xxx","app":"admin_audit","message":"Login successful: \"me\"","level":1,"time":"2016-07-03T20:22:01+02:00","method":"PUT","url":"\/remote.php\/webdav\/pics\/IMG_20160703_202123.jpg-chunking-1892-4-0","user":"me"}
{"reqId":"hWI7HcWyXiP8eWD9xF\/L","remoteAddr":"xxx.xxx.xxx.xxx","app":"admin_audit","message":"Login attempt: \"me\"","level":1,"time":"2016-07-03T20:22:01+02:00","method":"PUT","url":"\/remote.php\/webdav\/pics\/IMG_20160703_202123.jpg-chunking-1892-4-1","user":"--"}
{"reqId":"hWI7HcWyXiP8eWD9xF\/L","remoteAddr":"xxx.xxx.xxx.xxx","app":"admin_audit","message":"Login successful: \"me\"","level":1,"time":"2016-07-03T20:22:01+02:00","method":"PUT","url":"\/remote.php\/webdav\/pics\/IMG_20160703_202123.jpg-chunking-1892-4-1","user":"me"}
{"reqId":"2rDgQrQFCgkGCN3PAJQM","remoteAddr":"xxx.xxx.xxx.xxx","app":"admin_audit","message":"Login attempt: \"me\"","level":1,"time":"2016-07-03T20:22:02+02:00","method":"PUT","url":"\/remote.php\/webdav\/pics\/IMG_20160703_202123.jpg-chunking-1892-4-2","user":"--"}
{"reqId":"2rDgQrQFCgkGCN3PAJQM","remoteAddr":"xxx.xxx.xxx.xxx","app":"admin_audit","message":"Login successful: \"me\"","level":1,"time":"2016-07-03T20:22:02+02:00","method":"PUT","url":"\/remote.php\/webdav\/pics\/IMG_20160703_202123.jpg-chunking-1892-4-2","user":"me"}
{"reqId":"2rDgQrQFCgkGCN3PAJQM","remoteAddr":"xxx.xxx.xxx.xxx","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 400 expected filesize 1024000 got 417792\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\BadRequest\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/file.php(103): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->createFileChunked(Resource id #40)\\n#1 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/directory.php(134): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->put(Resource id #40)\\n#2 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1036): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Directory->createFile('IMG_20160703_20...', Resource id #40)\\n#3 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(523): Sabre\\\\DAV\\\\Server->createFile('pics\\\/IMG_201607...', Resource id #40, NULL)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#7 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(55): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/var\\\/www\\\/nextcloud\\\/remote.php(138): require_once('\\\/var\\\/www\\\/nextcl...')\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/file.php\",\"Line\":404,\"User\":\"me\"}","level":4,"time":"2016-07-03T20:22:02+02:00","method":"PUT","url":"\/remote.php\/webdav\/pics\/IMG_20160703_202123.jpg-chunking-1892-4-2","user":"me"}
```

**Comment by AndyScherzinger** [comment]
@aronovgj does this also happen with the current stable version 1.0.1? Just asking to see if this is a blocker for the release.

cc @MorrisJobke just to check if this could be server related.

**Comment by aronovgj** [comment]
Totally weird. So I uninstalled the RC and installed the stable but I couldn't reproduce the issue. Then I reinstalled the RC again and I'm still getting this:

```
{"reqId":"4DU3hZamygy+pylqQWjl","remoteAddr":"xxx.xxx.xxx.xxx","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 404 File with name pics\\\/IMG_20160704_001135.jpg could not be located\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\NotFound\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(79): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\ObjectTree->getNodeForPath('pics\\\/IMG_201607...', 0)\\n#1 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpGet(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#2 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#3 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:GET', Array)\\n#4 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(251): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response), false)\\n#5 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpHead(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#6 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#7 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:HEAD', Array)\\n#8 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#9 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(55): Sabre\\\\DAV\\\\Server->exec()\\n#10 \\\/var\\\/www\\\/nextcloud\\\/remote.php(138): require_once('\\\/var\\\/www\\\/nextcl...')\\n#11 {main}\",\"File\":\"\\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/connector\\\/sabre\\\/objecttree.php\",\"Line\":168,\"User\":\"me\"}","level":0,"time":"2016-07-04T00:12:37+02:00","method":"HEAD","url":"\/remote.php\/webdav\/pics\/IMG_20160704_001135.jpg","user":"me"}
```

But the uploads are not failing anymore. I took like a hundred pictures and it didn't fail even once while before it happened about every second or third time.

I'll report back if the issue returns...

**Comment by AndyScherzinger** [comment]
Weird... @aronovgj should we close the issue then for now? The 404 could maybe, not sure, who knows be a database problem/hiccup since RC2 you can wipe the app's database (Android settings->Apps->Nextcloud->Manage...) which might help but will also remove any account etc. configured within the app.

**Comment by aronovgj** [comment]
Yes, let's close this for now

**Comment by aronovgj** [comment]
It happens again. Now only 400 errors.
`{\"Message\":\"HTTP\\\/1.1 400 expected filesize...`

**Comment by bong0** [comment]
I'm having this issue with 1.3.0 RC2 (play store beta). I cannot provide you with server log since it's hosted by a third party. A client log would be helpful but seems not to be implemented.

**Comment by AndyScherzinger** [comment]
@bong0 the client log is only activated in the beta app which can be installed in parallel and should give you a client. Would you mind trying to reproduce this issue with the beta app? That would help us a lot :) Thanks in advance!

**Comment by bong0** [comment]
@AndyScherzinger I have the beta app installed but there is no note given in it where I should find that log.

**Comment by AndyScherzinger** [comment]
the log is at the bottom in the settings screen :)

**Comment by bong0** [comment]
Unfortunately not (see images)

Am 13. September 2016 21:36:36 MESZ, schrieb Andy Scherzinger <notifications@github.com>:
>the log is at the bottom in the settings screen :)
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/nextcloud/android/issues/85#issuecomment-246798238

-- 
Gesendet von meinem Mobiltelefon

**Comment by bong0** [comment]
Unfortunately not 
![screenshot_20160913-215452](https://cloud.githubusercontent.com/assets/179111/18489293/412c5934-79fd-11e6-8bcb-d0115351539c.png)
![screenshot_20160913-215521](https://cloud.githubusercontent.com/assets/179111/18489294/4147ff9a-79fd-11e6-8023-c0831099bcd1.png)

**Comment by AndyScherzinger** [comment]
@bong0 my mistake sorry, should have posted the link... the "real" beta is only available on f-droid... https://f-droid.org/repository/browse/?fdfilter=com.nextcloud&fdid=com.nextcloud.android.beta and is rather for testing purposes but has a log for that matter :)

**Comment by bong0** [comment]
@AndyScherzinger this confusion should be definitely removed.

**Comment by AndyScherzinger** [comment]
@bong0 that is a tricky one I must say since:

* to get the release candidates from the Play store you register as a "tester" to recieve what they call a "beta"

The beta isn't promoted (too much) by us, so that there is always a critical mass to test drive the beta and provide us with feedback but still treats it as a beta/testing application so most communication is done via the discourse forum and the issue tracker. While the release candidates (the one you have been using before) is distributed via Google Play and is used to give people the change to test a feature complete release in their own/custom environment before we publish it for the larger community of users.

The differences are described in the documentation: https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md

**Comment by janvonde** [comment]
the same error happens to me ... nc 10 stable and android client beta via google play

**Comment by MrOerni** [comment]
Same here NC 10 stable, but the regular playstore version (not beta)

Wiped Cache -> Nothing

Upload via share from gallary -> working fine

```
{"reqId":"G77zFNkZEVinaYLjlPuV","remoteAddr":"xxx.xxx.xxx.xxx","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 400 expected filesize 1024000 got 770048\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\BadRequest\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php(104): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->createFileChunked(Resource id #87)\\n#1 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/Directory.php(137): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->put(Resource id #87)\\n#2 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1036): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Directory->createFile('IMG_20160918_22...', Resource id #87)\\n#3 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(523): Sabre\\\\DAV\\\\Server->createFile('SofortUpload\\\/20...', Resource id #87, NULL)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#7 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(60): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/var\\\/www\\\/nextcloud\\\/remote.php(165): require_once('\\\/var\\\/www\\\/nextcl...')\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php\",\"Line\":405,\"User\":\"XXX\"}","level":4,"time":"2016-09-19T00:51:50+02:00","method":"PUT","url":"\/remote.php\/webdav\/SofortUpload\/2016\/09\/IMG_20160918_224918.jpg-chunking-11477953-4-3","user":"XXX"}
```

**Comment by aronovgj** [comment]
Are you guys having this problem by any chance using a repeater? And if so, is this a fritzbox repeater?

**Comment by MrOerni** [comment]
@aronovgj No

**Comment by bong0** [comment]
@aronovgj I don't use a repeater and have no access to it.

And the upload error also appears in the fdroid beta for me. I still cannot provide you with a log since the app did not generate any (it's empty)

I discovered another bug though when trying to send the empty log via the share intent in nextcloud (attached) 
[ocbetaerr.txt](https://github.com/nextcloud/android/files/484467/ocbetaerr.txt)

**Comment by AndyScherzinger** [comment]
@bong0 for your bug I opened a new issue - #283

**Comment by Organized92** [comment]
I also have this bug, auto-upload gives me "Unknown error" on upload (not always, but often). Uploading files manually is working fine.

In my server log, there are lines like these:
`{"reqId":"\/INWrncGKPYG\/FBKoZrF","remoteAddr":"84.118.175.240","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 404 File with name Handybilder\\\/2016\\\/09\\\/IMG_20160921_224039.jpg could not be located\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\NotFound\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(79): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\ObjectTree->getNodeForPath('Handybilder\\\/201...', 0)\\n#1 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpGet(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#2 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#3 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:GET', Array)\\n#4 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(251): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response), false)\\n#5 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpHead(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#6 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#7 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:HEAD', Array)\\n#8 \\\/var\\\/www\\\/nextcloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#9 \\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(60): Sabre\\\\DAV\\\\Server->exec()\\n#10 \\\/var\\\/www\\\/nextcloud\\\/remote.php(165): require_once('\\\/var\\\/www\\\/nextcl...')\\n#11 {main}\",\"File\":\"\\\/var\\\/www\\\/nextcloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/ObjectTree.php\",\"Line\":173,\"User\":\"Andy\"}","level":0,"time":"2016-09-21T20:45:41+00:00","method":"HEAD","url":"\/remote.php\/webdav\/Handybilder\/2016\/09\/IMG_20160921_224039.jpg","user":"Andy"}`

Even if I try to restart the upload, it's not working. Latest android client version from the Play Store and the lates NC version.

**Comment by cestmoimeme** [comment]
save problem too ...
{"reqId":"GbIp8uRx2tzuTZsyXwel","remoteAddr":"82.243.93.115","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 400 expected filesize 10000000 got 9912320\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\BadRequest\",\"Code\":0,\"Trace\": etc etc

**Comment by AndyScherzinger** [comment]
@cestmoimeme mac client is handled in https://github.com/nextcloud/client_theming/issues cc @LukasReschke

**Comment by cestmoimeme** [comment]
ok, thnak you, and sorry :)

**Comment by AndyScherzinger** [comment]
@cestmoimeme no worries, everything is fine! There is an issue present and youo took the time to report it, so thank you for taking the time! :)

**Comment by blue1919** [comment]
i had the same issue with instantupload in until i did the following:
1. freed lots of space on my samsung note4.
2. removed the app and all traces of its cache
3. reinstalled the app.

its workibg like a charm again.. i knew it wasnt a server issue cz i have other users on this server. 

phone is android marshmallow with device encryption turned on.

hope this helps someone out there.

will report here again if problem comes back.

**Comment by aronovgj** [comment]
I have an idea why that happens. As I mentioned, I have a oneplus 3. The camera takes a raw picture and what I think happens then is that this picture is saved to the picture directory, post-processed and replaced. So the app tries to upload the picture which is replaced by a new one in the middle of the upload. That's why it fails. I think. 

For some reason though, I can not re-upload the picture if it failed once. It only happens with the stock app. With other apps it works well but those don't do post-processing. At least not the way the stock app does it.

**Comment by aronovgj** [comment]
Would probably help if someone with this issue tried to reproduce it with snap camera as I don't have those issues using that app. (not a solution for me because the quality is worse)

**Comment by aronovgj** [comment]
I updated to OOS 4.0 (Nougat) and the issue seems to have disappeared.

**Comment by rrjk** [comment]
I've been having the same issues. Sometimes the auto-upload succeeds (rarely), typically it fails with logs like in the previous messages.
I've not yet updated to Nougat and based on the comment of aronovgj I've tried installing Open Camera. 
With Open Camera the auto upload now works (I've taken a few pictures) fine.

For photos that failed upload, I also experience that sometimes a retry does work and sometimes it doesn't. I see similar behavior when I try to share the photo with the Nextcloud app directly from the Gallery.

**Comment by aronovgj** [comment]
I suspect that this is not connected to Nougat but to changes done to the camera firmware by Oneplus.

**Comment by tobiasKaminsky** [comment]
In our latest beta app @mario did real great enhancements to the autoupload.
Please try it either via fdroid or install it https://github.com/nextcloud/android/blob/beta/apks/latest.apk manually.

**Comment by aronovgj** [comment]
Seems to be obsolete now.
